### PR TITLE
feat: Add experimental image API

### PR DESCRIPTION
**Pull request by mrkurt**
`fly.Image` now allows image conversion and resize
operations. Supported functions:

* `resize`: http://sharp.pixelplumbing.com/en/stable/api-resize/#resize
* `crop`: http://sharp.pixelplumbing.com/en/stable/api-resize/#crop
* `embed`: http://sharp.pixelplumbing.com/en/stable/api-resize/#embed
* `webp`: http://sharp.pixelplumbing.com/en/stable/api-output/#webp
* `png`: http://sharp.pixelplumbing.com/en/stable/api-output/#png

Run `toBuffer` to get a raw `ArrayBuffer` from chained operations.

### CODE REVIEW

File: src/bridge/fly/image.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          throw new Error("Invalid image operation: " + o.name)
+      }
+    }
+  }catch(err){
+    ctx.tryCallback(callback, [err.toString()])
+    return 
+  }
+
+  const inSize = data.byteLength
+  //log.debug(image)
```
Comment (jeromegn) [inline src/bridge/fly/image.ts]: Remove this line? Maybe all the other log.debug, unless you still need them.

File: v8env/fly/index.js [hunk 2]
```diff
@@ -16,6 +17,9 @@ export default function flyInit(ivm, dispatcher) {
         hash: SparkMD5.hash
       }
     },
-    streams: streamsInit(ivm, dispatcher)
+    streams: streamsInit(ivm, dispatcher),
+    experimental: {
+      Image: initImage(ivm, dispatcher)
```
Comment (jeromegn) [inline v8env/fly/index.js]: `streamsInit` and `initImage` :) which format should we use?

File: package.json [hunk 3]
```diff
@@ -71,6 +74,7 @@
     "whatwg-url": "^6.4.0"
   },
   "dependencies": {
+    "@types/sharp": "^0.17.6",
```
[the remark this answers is not in the document]
Comment (mrkurt) [reply package.json]: It needs this to run, not just for v8.

[the remark this answers is not in the document]
Comment (jeromegn) [reply package.json]: Oh yes, right. I see that now.

**Comment by jeromegn** [comment]
I have an additional comment: What's the point of implementing `embed` without `background`? I looked at the docs and embed only seems useful if there's a background. I'm a bit confused by `embed` in any case, care to enlighten? :)

**Comment by mrkurt** [comment]
Oh, derp, you're right.

Embed is how you resize an image that's, like, 600x600 down to a 300x200
without cropping it. It fills the dead space with a background color (with
alpha).

**Pull request merged**